pkg/leakybucket/reset_filter.go

```diff
@@ -0,0 +1,66 @@
+package leakybucket
+
+import (
+	"time"
+
+	"github.com/antonmedv/expr"
+	"github.com/antonmedv/expr/vm"
+
+	"github.com/crowdsecurity/crowdsec/pkg/exprhelpers"
+	"github.com/crowdsecurity/crowdsec/pkg/types"
+)
+
+// ResetFilter allows to kill the bucket (without overflowing), if a particular condition is met.
+// An example would be a scenario to detect aggressive crawlers that *do not* fetch any static ressources :
+// type : leaky
+// filter: filter: "evt.Meta.log_type == 'http_access-log'
+// reset_filter: evt.Parsed.request endswith '.css'
+// ....
+// Thus, if the bucket receives a request that matches fetching a static ressource (here css), it cancels itself
+
+type ResetFilter struct {
+	ResetFilter          *vm.Program
+	ResetFilterSafeGuard map[string]time.Time
+}
```

Comment:
Do we want also an `ExprDebugger` for this ?

Reply:
@AlteredCoder isnt `ExportDebugger` attached AT run/compile time?